### PR TITLE
Added test for parent-child-relation

### PR DESCRIPTION
**Pull request by Davknapp**
I implemented a test, that checks if the parent of an element was computed correctly

### CODE REVIEW

File: test/t8_test_find_parent.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    /*Compute parent of child */
+    ts->t8_element_parent (child, test_parent);
+    /*If its equal, call child_find_parent, to check if parent-child relation
+     * is correct in next level until maxlvl is reached*/
+    if (ts->t8_element_compare (element, test_parent)) {
+      SC_ABORT ("Computed child_parent is not the parent");
+    }
+    else {
+      t8_recursive_child_find_parent (child, ts, level + 1, maxlvl);
+    }
```
Comment (holke) [inline test/t8_test_find_parent.cpp]:
```suggestion
    SC_CHECK_ABORT (!ts->t8_element_compare (element, test_parent), "Computed child_parent is not the parent");
    t8_recursive_child_find_parent (child, ts, level + 1, maxlvl);
```

We use SC_CHECK_ABORT to check for a condition and abort with a message if the condition is not met.
You can also use SC_CHECK_ABORTF for formatted output in the message.

File: test/t8_test_find_parent.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+{
+  T8_ASSERT (level <= maxlvl
+             || maxlvl < ts->t8_element_num_children (element));
+  int                 num_children, i;
+  t8_element_t       *child, *test_parent;
+  /*Get number of children */
+  num_children = ts->t8_element_num_children (element);
+  /*Get child and test_parent, to check if test_parent = parent of child */
+  ts->t8_element_new (1, &child);
+  ts->t8_element_new (1, &test_parent);
```
[the remark this answers is not in the document]
Comment (holke) [reply test/t8_test_find_parent.cpp]: Also make sure that child and test_parent are destroyed again after use.
Currently the memory is not free'd

File: test/t8_test_find_parent.cpp [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+{
+  t8_element_t       *element;
+  t8_scheme_cxx      *scheme;
+  t8_eclass_scheme_c *ts;
+  int                 eclassi;
+  t8_eclass_t         eclass;
+  scheme = t8_scheme_new_default_cxx ();
+  for (eclassi = T8_ECLASS_ZERO; eclassi < T8_ECLASS_PYRAMID; eclassi++) {
+    eclass = (t8_eclass_t) eclassi;
+    /*Get scheme for eclass */
```
Comment (holke) [inline test/t8_test_find_parent.cpp]:
```suggestion
    /* Get scheme for eclass */
```

File: test/t8_test_find_parent.cpp [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    ts = scheme->eclass_schemes[eclass];
+    /*Get element and initialize it */
```
Comment (holke) [inline test/t8_test_find_parent.cpp]:
```suggestion
    /* Get element and initialize it */
```

File: test/t8_test_find_parent.cpp [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    ts->t8_element_new (1, &element);
+    ts->t8_element_set_linear_id (element, 0, 0);
+    /*Check for correct parent-child relation */
```
Comment (holke) [inline test/t8_test_find_parent.cpp]:
```suggestion
    /* Check for correct parent-child relation */
```

File: test/t8_test_find_parent.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Davknapp) [reply test/t8_test_find_parent.cpp:45]: > Allocate these outside of the recursion and reuse them.
> This way you allocate more elements on each level, which should not be necessary.

I think I can reuse the child, but must store somehow the parent. 
I guess, you have something in mind like
parent = child,
t8_recursive_child_find_parent(child, parent ....)?
But in that way, if we go up again in the recursion, the parent is overwritten and the next computed child is computed at the deepest level, not the next level.

**Comment by holke** [comment]
And right when you think you are done, the tests fail.
Looks to me like we are out of memory in the non-debug case, but i'm just guessing.
We need to investigate further why it crashes now.

**Pull request merged**